src/bashDebug.ts

```diff
@@ -71,32 +77,73 @@ export class BashDebugSession extends LoggingDebugSession {
 			this.sendResponse(response)
 		});
 
-		spawn("bash", ["-c", `pkill -KILL -P ${this.debuggerProcessParentId}`]);
+		spawn("bash", ["-c", `${this.launchArgs.pathPkill} -KILL -P ${this.debuggerProcessParentId}`]);
+	}
+
+	/**
+	 * TODO: replace return type with something like errno in POSIX.
+	 */
+	private isPathValid(args: LaunchRequestArguments): boolean {
+		const argv = ["-c",
+			`type ${args.pathBashdb} || exit 1;` +
+			`type ${args.pathCat} || exit 2;` +
+			`type ${args.pathMkfifo} || exit 3;` +
+			`type ${args.pathPkill} || exit 4;` +
+			`test -d ${args.cwd} || exit 11;` +
+			""
+		]
+		const proc = spawnSync(args.pathBash, argv, { timeout: 1000 });
+
+		if (proc.error !== undefined) {
+			// @ts-ignore Property 'code' does not exist on type 'Error'.
+			if (proc.error.code === "ENOENT") {
+				// bash not found
+				return false;
+			}
+			// @ts-ignore Property 'code' does not exist on type 'Error'.
+			if (proc.error.code === "ETIMEDOUT") {
+				// TODO: It's possibly bug. Notify it to user.
+				return false;
+			}
+			// TODO: Else, unknown error. Show dialog to promote user to report the issue.
+			return false;
+		}
+
+		if (proc.status != 0) {
+			return false;
+		}
+
+		return true;
```

Comment:
I had problem here with "C:/windows/System32/bash.exe" not found, while this works "C:\\windows\\sysnative\\bash.exe". 

Seems simple "Bash Debug: launch failed" is too generic.

I propose something alike:
![image](https://user-images.githubusercontent.com/23117003/37245591-be93580c-249a-11e8-83c5-0352ae5c7ebc.png)

```javascript
	private vaalidatePath(args: LaunchRequestArguments): string | undefined {
		const argv = ["-c",
			`type ${args.pathBashdb} || exit 1;` +
			`type ${args.pathCat} || exit 2;` +
			`type ${args.pathMkfifo} || exit 3;` +
			`type ${args.pathPkill} || exit 4;` +
			`test -d ${args.cwd} || exit 11;` +
			""
		]
		const proc = spawnSync(args.pathBash, argv, { timeout: 1000 });

		if (proc.error !== undefined) {
			return `${proc.error}`;
		}

		if (proc.status != 0) {
			return `${proc.status}`;
		}

		return undefined;
	}

...

		var pathValidationError = this.vaalidatePath(args);
		if (pathValidationError !=  undefined) {
			response.success = false;
			response.message = `Bash Debug: launch failed with ${pathValidationError}`;
			this.sendResponse(response);
			return;
		}
````
or similar...